### PR TITLE
CHECKOUT-6781: Set up local plugin and generator for workspace

### PR DESCRIPTION
**Pull request by davidchin**
## What?
* Convert `auto-export` build script into a Nx generator.
* Create `workspace-tools`, local Nx plugin, to host `auto-export` generator, as well other generators or executors required by this workspace in the future.

## Why?
Currently the `auto-export` build script is executed as a part of Webpack build process. Considering Nx has now been set up, a better way to execute the task is via Nx instead, as it can better leverage build cache.

## Testing / Proof
CircleCI

@bigcommerce/checkout

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -144,7 +145,7 @@
     "ts-jest": "^25.5.1",
     "ts-loader": "^9.3.0",
     "tslib": "^1.10.0",
-    "typescript": "^3.9.7",
+    "typescript": "4.1.2",
```
Comment (bc-peng) [inline package.json:148]: 😃 Thanks.

File: packages/workspace-tools/.eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    },
+    {
+      "files": [
+        "*.js",
+        "*.jsx"
+      ],
+      "rules": {}
+    }
+  ]
+}
```
Comment (bc-peng) [inline packages/workspace-tools/.eslintrc.json:33]:
```suggestion
{
  "extends": ["../../.eslintrc.json"]
}
```
Think we can simplify it as a result of my last week's update to `../../.eslintrc.json`.

Comment (davidchin) [reply packages/workspace-tools/.eslintrc.json:33]: Ah nice one, thanks! 👍

**Review by bc-peng — APPROVED** [review]
LGTM, thanks.

**Pull request merged**